### PR TITLE
docs: Update docs with advance usage

### PR DESCRIPTION
**Pull request by nadeesha**

### CODE REVIEW

**Review by johnjcsmith — APPROVED** [review]

File: docs/advanced/compute-recovery.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+Status: **General Availability**
+
+In a cloud environment, machines can fail at any time. Differential transparently handles machine failures, and retries the operation on a healthy worker. This means that you don't have to worry about your service being unavailable due to a machine failure.
+
+Machines periodically send heartbeats to the control-plane. If a machine fails to send any heartbeats within a certain interval (default 90 seconds):
+
+1. It is marked as unhealthy, and Differential will not send any new requests to it.
+2. The functions in progress are marked as failed, and Differential will retry them on a healthy worker.
+
+If the machine comes back online, Differential will mark it as healthy, and start sending new requests to it. However, it will disregard any results from the machine for the functions that were marked as failed.
```
Comment (ShawnSomething) [inline docs/advanced/compute-recovery.md:12]: When the machine comes back online

File: docs/advanced/distributed-caching.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
+# Distributed Caching
+
+Status: **General Availability**
+
+When a function returns a value, Differential will store it in the control-plane state. It's then available to be used by other calls, without having to call the function again. This is useful for caching expensive operations, and speeding up your application.
```
Comment (ShawnSomething) [inline docs/advanced/distributed-caching.md:5]: Can we swap the two sentences?

Differential is useful for caching expensive operations, and speeding up your application by storing returned function values in the control-plane. It is then available to be used by other calls without having to call that specific function again


(Value + How) : also it flows better with the next part. cause currently, it's how, then value, then the next paragraph is how again

File: docs/advanced/predictive-alerting.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
+# Predictive Alerting
+
+Status: **In Development**
+
+Differential can predict if a function is failing due to an unrecoverable error that requires a code change, and alert you with reproduction steps.
+
+The control-plane has a lot of context on a particular failure when it happens. It knows the function, some metadata about the source, payload, and the error message. It can use this information to predict if the error is unrecoverable or not.
```
Comment (ShawnSomething) [inline docs/advanced/predictive-alerting.md:7]: The control-plane has all the required context on a particular failure, when it happens.

File: docs/advanced/predictive-retries.md [hunk 4]
```diff
@@ -0,0 +1,42 @@
+# Predictive Retries
+
+Status: **Technical Preview**
+
+Differential can predict transient errors and retry the operations without the developer having to write custom code.
+
+The control-plane has a lot of context on a particular failure when it happens. It knows the function, some metadata about the source, payload, and the error message. It can use this information to predict if the error is transient or not.
```
Comment (ShawnSomething) [inline docs/advanced/predictive-retries.md:7]: has all the required context

**Review by ShawnSomething — COMMENTED** [review]
a couple of suggestions for reading flow

File: docs/advanced/compute-recovery.md (hunk 1 above)
Comment (nadeesha) [reply docs/advanced/compute-recovery.md:12]: I think it's `If`. Sometimes the machine doesn't come online.

**Pull request merged**